### PR TITLE
CMake install

### PR DESCRIPTION
**Pull request by krakjoe**
As requested by #116

### CODE REVIEW

**Comment by philip** [comment]
@krakjoe  It appears `uitable.h` should be removed from this patch as it was moved to a `table` branch.

File: CMakeLists.txt [hunk 1]
```diff
@@ -217,3 +217,15 @@ macro(_add_exec _name)
 endmacro()
 add_subdirectory("test")
 add_subdirectory("examples")
+
+install(TARGETS ${_LIBUINAME}
+   LIBRARY DESTINATION lib
+   ARCHIVE DESTINATION lib)
+install(FILES ui.h uitable.h DESTINATION include)
```
Comment (rwols) [inline CMakeLists.txt]: Do we still need `uitable.h`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -217,3 +217,15 @@ macro(_add_exec _name)
...
+if (APPLE)
+	install(FILES ui_darwin.h DESTINATION include)
+elseif (WIN32)
+	install(FILES ui_windows.h DESTINATION include)
+else()
+	install(FILES ui_unix.h DESTINATION include)
+endif()
```
Comment (rwols) [inline CMakeLists.txt]: You can just do `install(FILES ui.h ui_${_OSNAME}.h DESTINATION include)`, no?

In fact, wouldn't it be even better if we did `set_target_properties(${_LIBUINAME} PROPERTIES PUBLIC_HEADER ui.h ui_${_OSNAME}.h)`, and then `install(TARGETS ${_LIBUINAME} LIBRARY DESTINATION lib RUNTIME DESTINATION lib PUBLIC_HEADER DESTINATION include)`?

**Comment by andlabs** [comment]
No longer relevant given the upcoming (either tomorrow or Tuesday) switch to Meson. In fact, I've also set up the install rules in this new environment, so you finally get your install rules. Thanks anyway!

**Pull request closed without merging**